Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/AutoLift.java

```diff
@@ -12,103 +12,105 @@
 /**
  *
  */
-public class AutoLift extends Command implements PIDOutput{
+public class AutoLift extends Command implements PIDOutput {
 	/**
-	 * All distances are measured from bottom of cube and + 3 inches for wiggle room for dropping cubes
-	 * Also, actual distances are divided by 3 because according to cad, the lift will have a 1:3 ratio from winch
-	 * to actual height.
+	 * All distances are measured from bottom of cube and + 3 inches for wiggle room
+	 * for dropping cubes Also, actual distances are divided by 3 because according
+	 * to cad, the lift will have a 1:3 ratio from winch to actual height.
 	 */
-	
-	/**
-	 * Distance to switch
-	 * 18.75 inches in starting position (this measurement is the fence that surrounds the switch)
-	 * 21.75 / 3 for ratio = 7.25
-	 */
-	private final double SWITCH_DIST = 7.25;
-	/**
-	 * Distance to scale
-	 * 5 feet starting
-	 * 63 / 3 = 21
-	 */
-	private final double SCALE_DIST = 21;
-	/**
-	 * Distance to bar
-	 * 87 / 3 = 29
-	 * 7 feet starting; bar distance should be changed because I'm not aware how climber mech will be positioned
-	 */
-	private final double BAR_DIST = 29;
+
+	private final int NUM_STAGES;
+	private final double WIGGLE_ROOM;
+	private final double SWITCH_DIST;
+	private final double SCALE_DIST;
+	private final double BAR_DIST;
+
 	private double desiredDist = 0;
 	private double currDist;
 	private LiftInterface lift;
-	private Position desiredPos;
-	
+	private LiftHeight desiredPos;
+
 	private PIDController liftController;
 
-    public AutoLift(Position stage, LiftInterface lift) {
-    	this.lift = lift;
-    	requires(Robot.lift);
-    	currDist = lift.getHeight();
-    	switch(stage) {
-    	case GROUND:
-    		desiredDist = -currDist;
-    		break;
-    	case SWITCH:
-    		desiredDist = SWITCH_DIST - currDist;
-    		break;
-    	case SCALE:
-    		desiredDist = SCALE_DIST - currDist;
-    		break;
-    	case BAR:
-    		desiredDist = BAR_DIST - currDist;
-    		break;
-    	}
-    	
-    	liftController = new PIDController(Robot.getConst("LiftkP", 1), Robot.getConst("LiftkI", 0),
-				Robot.getConst("LiftkD", 0),RobotMap.liftEnc, this);
-    	
-    	desiredPos = stage;
-    }
-
-    // Called just before this Command runs the first time
-    protected void initialize() {
+	public AutoLift(LiftHeight stage, LiftInterface lift) {
+		this.lift = lift;
```

Comment:
Can't this command just use the methods of the Lift subsystem? Note that if you need access to it's PIDController, you can use [getPIDController](http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/command/PIDSubsystem.html#getPIDController--)

Reply:
Yes. It's kind of a mute point though: my intention was to not use this command at all and instead use LiftToPosition as the better, newest implementation